loki/subroutine.py

```diff
@@ -104,6 +104,10 @@ def __initialize__(
         self.result_name = result_name
         self.is_function = is_function
 
+        # Make sure 'result_name' is defined if it's a function
+        if self.result_name is None and self.is_function:
+            self.result_name = name
```

Comment:
Thanks! With this, we can now simplify the code here:
https://github.com/ecmwf-ifs/loki/blob/6b58fe7fef3e151bbf74df18e034a1bc7924e023/loki/subroutine.py#L330-L332

Would you mind replacing this simply with the following?
```py
return self.symbol_attrs.get(self.result_name) 
```